main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -1096,6 +1097,35 @@ object Parser2 {
       }
     }
 
+    private def structDecl(mark: Mark.Opened)(implicit s: State): Mark.Closed = {
+      assert(at(TokenKind.KeywordStruct))
+      expect(TokenKind.KeywordStruct, SyntacticContext.Decl.Struct)
+      val nameLoc = currentSourceLocation()
+      name(NAME_TYPE, context = SyntacticContext.Decl.Struct)
+      if (at(TokenKind.BracketL)) {
```

Comment:
Later, we will require all structs to have at least one type parameter (the region), but I suppose this is OK for now.